pkg/operator/controller/machineconfig/controller.go

```diff
@@ -265,7 +259,6 @@ func (r *MachineConfigReconciler) removeFinalizer(ctx context.Context, req ctrl.
 
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		if err := r.ClientGet(ctx, req.NamespacedName, withoutFinalizers); err != nil {
-			r.Log.Error(err, "failed to fetch nodeobservabilitymachineconfig resource for removing finalizer")
```

Comment:
Maybe we could add the following for context: 
```
err = utilerrors.NewAggregate([]error{err, fmt.Errorf("failed to fetch nodeobservabilitymachineconfig resource for removing finalize: %w", err)}
```

Reply:
This message doesn't really bring any new information: 
- if fetching fails it'll be in the error
- the fact that it's done for the removal of the finalizers will be mentioned up in the call stack (`cleanUp` method)